### DISCUSSION TITLE
How to Find if an Image is Animatable

### DISCUSSION THREAD

**Discussion by AmitSonkhiya** [libvips/php-vips#211] [Q&A] [answered]
Hi again,

I was trying to find a function in Libvips docs that could return image iterations as an integer.
Preciously, I would like to find if an image is animatable or not. Hence a boolean/BIT return value is sufficient as well.

PHP [getimageiterations()](https://www.php.net/manual/en/imagick.getimageiterations.php) function gives a BIT value for this purpose  and I would like to find if Libvips has a similar function for the same.

Is there any function in Libvips to find if an image is animatable, specially Webp images?

Thank you

PS: It is pretty hard to find functions for purposes.
I wish there could be a feature in the official documentation to search with related words.
I tried Google but it yield nothing.

**Comment by jcupitt** [comment]
Hello @AmitSonkhiya,

You need to look at the image metadata. For example:

```
$ vipsheader -a dancing_banana2.lossless.webp 
dancing_banana2.lossless.webp: 990x1050 uchar, 4 bands, srgb, webpload
width: 990
height: 1050
bands: 4
format: uchar
coding: none
interpretation: srgb
xoffset: 0
yoffset: 0
xres: 1
yres: 1
filename: dancing_banana2.lossless.webp
vips-loader: webpload
loop: 0
delay: 100 100 100 100 100 100 100 100 
n-pages: 8
```

In PHP you can use eg. `$image->get("n")` to get the number of frames in an animation.

> I wish there could be a feature in the official documentation to search with related words.

I usually use the function list:

https://www.libvips.org/API/current/func-list.html

**Reply by jcupitt** [reply]
There's also a "search" box at the top-right of the main PHP docs, of course:

https://libvips.github.io/php-vips/classes/Jcupitt-Vips-Image.html

**Reply by AmitSonkhiya** [reply]
Hello again @jcupitt 

Thanks for letting me know.

```
$image->getType('n');
Outputs: 0  
```
Is it `$image->get("n")` or `$image->get("n-pages")`?
Because with `n`, it gives error: vips_image_get: field "n" not found.
With `n-pages`, it gives a number. I guess it is the frame count.

Thank you

About search, yep! those tools are useful.
Just I was searching on Google with human language terms like "Libvips/PHP vips check if an image is animatable", with a hope that it would lead me to some GitHub discussion or issue to find a clue. Hehe!!

**Reply by jcupitt** [reply, marked as answer]
Oop, `n-pages`, of course you're right.
